pkg/scheduler/algorithm/priorities/metadata.go

```diff
@@ -67,7 +79,24 @@ func (pmf *PriorityMetadataFactory) PriorityMetadata(pod *v1.Pod, nodeNameToInfo
 		podSelectors:            getSelectors(pod, pmf.serviceLister, pmf.controllerLister, pmf.replicaSetLister, pmf.statefulSetLister),
 		controllerRef:           priorityutil.GetControllerRef(pod),
 		podFirstServiceSelector: getFirstServiceSelector(pod, pmf.serviceLister),
+		nodeUtilInfo:            getNodeUtilizationInfo(pmf.metricsClient),
```

Comment:
This is in the critical path of scheduling. Going over the network to get information could cause significant delay. If the server is unresponsive or slow, it can be a lot worse. Instead, we should probably have a cache to store metrics and have a separate goroutine that updates the cache frequently. In the critical path of scheduling, we only refer to the cache and never go directly to the metrics server.